Makefile

```diff
@@ -149,7 +151,12 @@ endif
 	touch $@
 
 lint: .gopathok varlink_generate ## Execute the source code linter
+ifeq ($(PRE_COMMIT),)
+	$(warning pre-commit was not found, we will try to install it.)
+	$(PYTHON) -m pip install --user pre-commit
```

Comment:
I'm sorry, I don't think I'm explaining myself properly.

As a developer, I do not want surprises from `make`. I do not want my home directory cluttered, my system altered without my consent or even awareness. This is a violation of trust.

You are welcome to issue a failure message telling the user what the problem is and how to solve it:
```
    FATAL: pre-commit tool not found. To continue, please run $(PYTHON) -m pip install --user pre-commit
```
You are welcome to add a new `install.precommit` target and encourage the user to use that. (You are NOT welcome to add that to `install.tools`, because that is so far entirely go-based and mostly required for actual building. This new thing is Python and not required for building, and as such deserves to be kept separate).

I would also encourage you to add these steps after `.tool/lint`: that way a developer at least gets the benefit of that linter even if they choose not to install pre-commit.

I hope this clarifies my position.